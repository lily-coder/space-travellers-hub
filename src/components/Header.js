import React from 'react';
import Navbar from './Navbar.js';

const Header = () => (
  <header className="header-section">
    <Navbar />
  </header>
);

export default Header;
