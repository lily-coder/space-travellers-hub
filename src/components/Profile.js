/* eslint no-unused-vars : "off" */
import React from 'react';

function Profile() {
  return (
    <div>
      My Profile
    </div>
  );
}

export default Profile;