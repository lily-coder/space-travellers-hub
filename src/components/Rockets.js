/* eslint no-unused-vars : "off" */
import React from 'react';
import RocketsCards from '../pages/RocketsCards.js';

const Rockets = () => {
  const r = '';
  return (
    <div>
      <RocketsCards />
    </div>
  );
};

export default Rockets;