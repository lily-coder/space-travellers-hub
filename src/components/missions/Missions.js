/* eslint no-unused-vars : "off" */
import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import HeaderMission from './HeaderMission.js';
import BodyMission from './BodyMission.js';

const Missions = () => {
  const { missions } = useSelector((state) => state).missions;

  const dispatch = useDispatch();

  return (
    <section>
      <table>
        <HeaderMission />
        < BodyMission />
      </table>
    </section>
  );
};

export default Missions;