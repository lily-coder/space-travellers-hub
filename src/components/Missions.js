/* eslint-disable no-unused-vars */
import { React, useEffect } from 'react';
import { useDispatch } from 'react-redux';
import MissionsList from '../pages/MissionList.js';
import { fetchDataRequest } from '../redux/missions/missions.js';

const Missions = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchDataRequest());
  }, []);
  return (
    <div>
      <MissionsList />
    </div>
  );
};
export default Missions;