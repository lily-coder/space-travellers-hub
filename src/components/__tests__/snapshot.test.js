/* eslint-disable no-unused-vars */
import TestRenderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import {
  BrowserRouter as Router,
} from 'react-router-dom';
import Rockets from '../Rockets.js';
import Missions from '../Missions.js';
import Profile from '../Profile.js';
import Header from '../Header.js';
import store from '../../redux/configStore.js';

describe('Render the rockets list', () => {
  it('Renders the rockets in the rockets page', () => {
    const rockets = TestRenderer.create(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    ).toJSON();
    expect(rockets).toMatchSnapshot();
  });
});

describe('Render the missions list', () => {
  it('Renders the Missions in the missions page', () => {
    const missions = TestRenderer.create(
      <Provider store={store}>
        <Missions />
      </Provider>,
    ).toJSON();
    expect(missions).toMatchSnapshot();
  });
});

describe('Render the missions lists', () => {
  it('Renders the rockets and missions in profile page', () => {
    const profile = TestRenderer.create(
      <Provider store={store}>
        <Profile />
      </Provider>,
    ).toJSON();
    expect(profile).toMatchSnapshot();
  });
});

describe('Render the Navbar', () => {
  it('Renders the navbar', () => {
    const header = TestRenderer.create(
      <Provider store={store}>
        <Router>
          <Header />
        </Router>
      </Provider>,
    ).toJSON();
    expect(header).toMatchSnapshot();
  });
});