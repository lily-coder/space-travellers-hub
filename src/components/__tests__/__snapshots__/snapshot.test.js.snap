// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render the Navbar Renders the navbar 1`] = `
<header
  className="header-section"
>
  <nav
    className="header-navbar"
  >
    <div
      className="logo"
    >
      <img
        alt=""
        className="header-logo"
        height="50"
        src="logo.png"
        width="50"
      />
      
      <p
        className="space-travellers"
      >
        Space Traveler's Hub
      </p>
    </div>
    <div
      className="header-nav"
    >
      <ul
        className="header-nav_ul"
      >
        <li
          className="rockets"
        >
          <a
            aria-current="page"
            className="active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Rockets
          </a>
        </li>
        <li
          className="missions"
        >
          <a
            aria-current={null}
            href="/Missions"
            onClick={[Function]}
          >
            Missions
          </a>
        </li>
        <li
          className="profile"
        >
          <a
            aria-current={null}
            href="/Profile"
            onClick={[Function]}
          >
            My Profile
          </a>
        </li>
      </ul>
    </div>
  </nav>
</header>
`;

exports[`Render the missions list Renders the Missions in the missions page 1`] = `null`;

exports[`Render the missions lists Renders the rockets and missions in profile page 1`] = `
<div>
  <div
    className="profile-section"
  >
    <div
      className="my-missions-profile"
    >
      <h2
        className="profile-headers"
      >
        My Missions
      </h2>
      <div>
        <ul />
      </div>
    </div>
    <div
      className="my-rockets-profile"
    >
      <h2
        className="profile-headers"
      >
        My Rockets
      </h2>
      <div>
        <ul />
      </div>
    </div>
  </div>
</div>
`;

exports[`Render the rockets list Renders the rockets in the rockets page 1`] = `
<div>
  <div
    className="rocket-card"
  />
</div>
`;
